src/compiler/manifest.rs

```diff
@@ -0,0 +1,633 @@
+// Copyright 2023 Mozilla Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The manifest is a description of all information needed to cache
```

Comment:
Can we rename `manifest`? It conflicts with the notion of `Cargo.toml` being referred to as _manifest_ which adds additional complexity to getting started with the codebase, especially in the `Storage` trait that is used across compiler domains.

Reply:
Sure, I've just used the `ccache` terminology, I agree we could use another name. `PreprocessorCacheEntry` could work. What do you think?